src/odemis/driver/simstreakcam.py

```diff
@@ -0,0 +1,537 @@
+        return md
+
+    def terminate(self):
+        try:
+            self._stop()  # stop any acquisition
+        except Exception:
+            pass
+
+    def _start(self):
+        """
+        Start an acquisition.
+        """
+        if self._generator is not None:
+            logging.warning("Generator already running")
+            return
+        self._generator = util.RepeatingTimer(self.exposureTime.value,
+                                              self._generate,
+                                              "SimCam image generator")
+        self._generator.start()
+
+    def _stop(self):
+        """
+        Stop the acquisition.
+        """
+        if self._generator is not None:
+            self._generator.cancel()
+            self._generator = None
+
+    def _getNewImage(self):
+        """
+        Gets a new image from the list of fake images provided.
+        :return: (dataarray) image
+        """
+        # image counter to provide different images to live view
+        if not hasattr(self, "_img_counter"):
+            self._img_counter = 0  # initialize the image counter
+        self._img_counter = self._img_counter % len(self._img_list)
+        image = self._img_list[self._img_counter]
+        self._img_counter += 1
+
+        return image
+
+    def _generate(self):
+        """
+        Generates the fake output image based on the resolution.
+        """
+        gen_img = self._getNewImage()
+        gen_img = self._simulate(gen_img)
+        timer = self._generator  # might be replaced by None afterwards, so keep a copy
+        self._waitSync()
+        if self._sync_event:
+            # If sync event, we need to simulate period after event (not efficient, but works)
+            time.sleep(self.exposureTime.value)
+
+        # update the trigger rate VA and MD for the current image
+        self.parent._delaybox._updateTriggerRate()
+
+        metadata = gen_img.metadata.copy()  # MD of image
+        metadata.update(self._metadata)  # MD of camera
+        self._mergeMetadata(metadata)
+
+        # update fake output metadata
+        exp = timer.period
+        metadata[model.MD_ACQ_DATE] = time.time() - exp
+        metadata[model.MD_EXP_TIME] = exp
+        metadata.pop(model.MD_PIXEL_SIZE, None)  # TODO get rid of this MD for now
+
+        # TODO check if that time list can be used to update the live view legend if there is one
+        if self.parent._streakunit.streakMode.value:  # if in Mode 'Operate'
+            # shape of gen_image is (time, lambda)
+            timeRange = self.parent._streakunit.timeRange.value  # time range of the time axis in image
+            # each px in time axis corresponds to a time stamp (easy approch: equally distributed)
+            step_time = timeRange/float(gen_img.shape[0])
+            metadata[model.MD_TIME_LIST] = list(numpy.arange(0, timeRange, step_time))  # append fake time list
```

Comment:
Use `numpy.linspace()`. With floats, it's safer to use (to ensure you obtain exactly the amount of elements you want), and more explicit too.